internal/services/compute/linux_virtual_machine_scale_set_resource.go

```diff
@@ -828,269 +826,273 @@ func resourceLinuxVirtualMachineScaleSetRead(d *pluginsdk.ResourceData, meta int
+						if err := d.Set("terminate_notification", FlattenVirtualMachineScaleSetScheduledEventsProfile(scheduleProfile)); err != nil {
+							return fmt.Errorf("setting `terminate_notification`: %+v", err)
+						}
+					}
+				}
 
-		extensionsTimeBudget := "PT1H30M"
-		if profile.ExtensionProfile != nil && profile.ExtensionProfile.ExtensionsTimeBudget != nil {
-			extensionsTimeBudget = *profile.ExtensionProfile.ExtensionsTimeBudget
-		}
-		d.Set("extensions_time_budget", extensionsTimeBudget)
+				if scheduleProfile := profile.ScheduledEventsProfile; scheduleProfile != nil {
+					if err := d.Set("termination_notification", FlattenVirtualMachineScaleSetScheduledEventsProfile(scheduleProfile)); err != nil {
+						return fmt.Errorf("setting `termination_notification`: %+v", err)
+					}
+				}
 
-		encryptionAtHostEnabled := false
-		vtpmEnabled := false
-		secureBootEnabled := false
+				extensionProfile, err := flattenVirtualMachineScaleSetExtensions(profile.ExtensionProfile, d)
+				if err != nil {
+					return fmt.Errorf("failed flattening `extension`: %+v", err)
+				}
+				d.Set("extension", extensionProfile)
 
-		if secprofile := profile.SecurityProfile; secprofile != nil {
-			if secprofile.EncryptionAtHost != nil {
-				encryptionAtHostEnabled = *secprofile.EncryptionAtHost
-			}
-			if uefi := profile.SecurityProfile.UefiSettings; uefi != nil {
-				if uefi.VTpmEnabled != nil {
-					vtpmEnabled = *uefi.VTpmEnabled
+				extensionsTimeBudget := "PT1H30M"
+				if profile.ExtensionProfile != nil && profile.ExtensionProfile.ExtensionsTimeBudget != nil {
+					extensionsTimeBudget = *profile.ExtensionProfile.ExtensionsTimeBudget
 				}
-				if uefi.SecureBootEnabled != nil {
-					secureBootEnabled = *uefi.SecureBootEnabled
+				d.Set("extensions_time_budget", extensionsTimeBudget)
+
+				encryptionAtHostEnabled := false
+				vtpmEnabled := false
+				secureBootEnabled := false
+
+				if secprofile := profile.SecurityProfile; secprofile != nil {
+					if secprofile.EncryptionAtHost != nil {
+						encryptionAtHostEnabled = *secprofile.EncryptionAtHost
+					}
+					if uefi := profile.SecurityProfile.UefiSettings; uefi != nil {
+						if uefi.VTpmEnabled != nil {
+							vtpmEnabled = *uefi.VTpmEnabled
+						}
+						if uefi.SecureBootEnabled != nil {
+							secureBootEnabled = *uefi.SecureBootEnabled
+						}
+					}
 				}
-			}
-		}
 
-		d.Set("encryption_at_host_enabled", encryptionAtHostEnabled)
-		d.Set("vtpm_enabled", vtpmEnabled)
-		d.Set("secure_boot_enabled", secureBootEnabled)
-		d.Set("user_data", profile.UserData)
-	}
+				d.Set("encryption_at_host_enabled", encryptionAtHostEnabled)
+				d.Set("vtpm_enabled", vtpmEnabled)
+				d.Set("secure_boot_enabled", secureBootEnabled)
 
-	if policy := props.UpgradePolicy; policy != nil {
-		d.Set("upgrade_mode", string(policy.Mode))
+				// userData is not returned by the API
+				d.Set("user_data", d.Get("user_data").(string))
```

Comment:
if we set the Option when retrieving the VMSS from the API then this can revert to:

```suggestion
				d.Set("user_data", profile.UserData)
```